metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -525,57 +527,40 @@ def get_datasource_urn(
-                    .get("native", {})
-                    .get("query", "")
-                )
-                parser = LineageRunner(raw_query)
-
-                for table in parser.source_tables:
-                    sources = str(table).split(".")
-
-                    source_db = sources[-3] if len(sources) > 2 else database_name
-                    source_schema, source_table = sources[-2], sources[-1]
-                    if source_schema == "<default>":
-                        source_schema = (
-                            database_schema
-                            if database_schema is not None
-                            else str(self.config.default_schema)
+                    name_components = [database_name, schema_name, table_name]
+                    return [
+                        builder.make_dataset_urn_with_platform_instance(
+                            platform=platform,
+                            name=".".join([v for v in name_components if v]),
+                            platform_instance=platform_instance,
+                            env=self.config.env,
                         )
-
-                    source_tables.add(
-                        f"{source_db + '.' if source_db else ''}{source_schema}.{source_table}"
-                    )
-            except Exception as e:
-                self.report.report_failure(
-                    key="metabase-query",
-                    reason=f"Unable to retrieve lineage from query. "
-                    f"Query: {raw_query} "
-                    f"Reason: {str(e)} ",
-                )
-                return None
-
-        if platform == "snowflake":
-            source_tables = set(i.lower() for i in source_tables)
-
-        # Create dataset URNs
-        dataset_urn = [
-            builder.make_dataset_urn_with_platform_instance(
+                    ]
+        else:
+            raw_query = (
+                card_details.get("dataset_query", {}).get("native", {}).get("query", "")
+            )
+            result = create_lineage_sql_parsed_result(
+                query=raw_query,
+                default_db=database_name,
+                default_schema=database_schema or self.config.default_schema,
                 platform=platform,
-                name=name,
                 platform_instance=platform_instance,
                 env=self.config.env,
+                graph=self.ctx.graph,
             )
-            for name in source_tables
-        ]
+            if result.debug_info.table_error:
+                logger.info(
+                    f"Failed to parse lineage from query {raw_query}: "
+                    f"{result.debug_info.table_error}"
+                )
+                self.report.report_warning(
+                    key="metabase-query",
+                    reason=f"Unable to retrieve lineage from query: {raw_query}",
+                )
+            return result.in_tables
```

Comment:
i guess we'll do CLL some other time?

Reply:
Didn't want to spend too much time on this. I'm also not sure how CLL is modeled on charts